src/Service/Resolvers/Sql Query Structures/BaseSqlQueryStructure.cs

```diff
@@ -41,10 +41,11 @@ public abstract class BaseSqlQueryStructure : BaseQueryStructure
         public string? FilterPredicates { get; set; }
 
         /// <summary>
-        /// DbPolicyPredicates is a string that represents the filter portion of our query
-        /// in the WHERE Clause added by virtue of the database policy.
+        /// Stores the database policies for all the constituent operations which get appended to the WHERE Clause.
+        /// For operations like PUT/PATCH, we may have 2 database policies, one for the update operation and the other for insert operation.
+        /// For simpler operations like GET, POST, DELETE, only one policy would be present.
         /// </summary>
-        public string? DbPolicyPredicates { get; set; }
+        public Dictionary<Config.Operation, string?> DbPolicyPredicatesForOperation { get; set; } = new();
```

Comment:
How does a single policy predicate string look like? 

What if the dbpolicy contains 2 policies with an && operator for eg? Does it have to be a list<string>?